tools/oeedger8r/Emitter.ml

```diff
@@ -637,21 +643,25 @@ let gen_enclave_code (ec : enclave_content) (ep : edger8r_params) =
     let oe_compute_buffer_size (buffer : string)
         (predicate : parameter_type * declarator -> bool) (plist : pdecl list)
         =
-      let rec gen_add_size prefix count (ptype, decl) =
+      let rec gen_add_size parent prefix count (ptype, decl) =
         let size = oe_get_param_size (ptype, decl, "_args." ^ prefix) in
         let arg = prefix ^ decl.identifier in
-        [ gen_c_for count
-        ; [sprintf "if (%s) OE_ADD_SIZE(%s, %s);" arg buffer size]
-        ; (let param_count =
-             oe_get_param_count (ptype, decl, "_args." ^ prefix)
-           in
-           flatten_map
-             (gen_add_size (arg ^ gen_c_deref param_count) param_count)
-             (get_deepcopy_members (get_param_atype ptype))) ]
-        |> List.flatten
+        let check_expr =
+          String.concat " && " (List.filter (( <> ) "") [parent; arg])
+        in
+        gen_c_for count
+          ( [ [sprintf "if (%s)" check_expr]
+            ; [sprintf "    OE_ADD_SIZE(%s, %s);" buffer size]
+            ; (let param_count =
+                 oe_get_param_count (ptype, decl, "_args." ^ prefix)
+               in
+               flatten_map
+                 (gen_add_size arg (arg ^ gen_c_deref param_count) param_count)
```

Comment:
Overall looks good. Can you double-check if we need to pass check_expr to the children?
Intuitively i would expect that check_expr gets passed down each level and each level adds another condition to the check_expr

Reply:
Yup we did, thank you! Fixed.